packages/grafana-ui/src/components/Legend/Legend.tsx

```diff
@@ -36,12 +36,18 @@ export type LegendPlacement = 'bottom' | 'right';
 
 export interface LegendOptions extends LegendBasicOptions, LegendRenderOptions {}
 
+export interface LegendSeriesRef {
+  fieldIndex: number;
+  frameIndex: number;
+}
```

Comment:
Maybe we could call this `FieldRef` or `DataFrameFieldRef` and move this to grafana/data?